opacus/accountants/utils.py

```diff
@@ -53,27 +51,40 @@ def get_noise_multiplier(
         raise NotImplementedError(
             "get_noise_multiplier is currently only supports RDP accountant"
         )
+    if (steps is None) == (epochs is None):
+        raise ValueError(
+            "get_noise_multiplier takes as input EITHER a number of steps or a number of epochs"
+        )
+    if steps is None:
+        steps = int(epochs / sample_rate)
 
     eps = float("inf")
-    sigma_min = DEFAULT_SIGMA_MIN_BOUND
-    sigma_max = DEFAULT_SIGMA_MAX_BOUND
     accountant = create_accountant(mechanism=accountant)
 
-    while eps > target_epsilon:
-        sigma_max = 2 * sigma_max
-        accountant.steps = [(sigma_max, sample_rate, int(epochs / sample_rate))]
-        eps = accountant.get_epsilon(delta=target_delta, **kwargs)
-        if sigma_max > MAX_SIGMA:
-            raise ValueError("The privacy budget is too low.")
+    accountant.steps = [(sigma_high, sample_rate, steps)]
+    eps_high = accountant.get_epsilon(delta=target_delta, **kwargs)
+    if eps_high > target_epsilon:
+        raise ValueError(
+            f"Noise multiplier of {sigma_high} is not enough to have a privacy budget of {target_epsilon}, please use a higher sigma_high."
+        )
+
+    accountant.steps = [(sigma_low, sample_rate, steps)]
+    eps = accountant.get_epsilon(delta=target_delta, **kwargs)
+    if eps < target_epsilon:
+        warnings.warn(
+            f"Noise multiplier of {sigma_low} is still too high for a privacy budget of {target_epsilon}, please use a lower sigma_low."
```

Comment:
I would replace "please use a lower sigma_low" with something like:
"you can benefit quality by allowing smaller sigma_low without exceeding target epsilon"

But the current version is also fine.